### PR TITLE
Fix vulnerability

### PR DESCRIPTION
**Pull request by ysugimoto**
I've noticed security vulnerability on this package, so tried to update mainly testing related package dependencies:

- nyc: 14.1.1
- ava: 2.0.0

And also bumped testing nodejs versions to `8.10` and `10.15`, the `6.10` is no longer supported in AWS, though. See: https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html

Then some tests always failed with `Test finished without running any assertions` I'm not figured out how tests are working. Any ideas? @kdybicz @1000ch

Test logs: https://travis-ci.org/ysugimoto/aws-lambda-image/jobs/542081018

### CODE REVIEW

File: test/s3-file-system.js [hunk 1]
```diff
@@ -27,7 +27,7 @@ test.before(async t => {
         }
     });
     AWS.mock( "S3", "putObject", (params, callback) => {
-        uploadedObject = params;
```
Comment (kdybicz) [inline test/s3-file-system.js]: all looks good, I just wonder why this change was needed. Could you shed some light on it?

Comment (ysugimoto) [reply test/s3-file-system.js]: Ah, I thought the reason for failing tests are due to asynchronous behavior for each `test` function, so every `putObject` params store to `uploadedObjects`.

Occasionally some assertions are processed to another test object, I thought (e.g testing file is `regular.jpg` but test seems to assert to `network-error.jpg`, maybe)

Comment (kdybicz) [reply test/s3-file-system.js]: sorry, I've missed that context. That totaly make sense.

**Comment by ysugimoto** [comment]
@kdybicz Yes, I'm wondering why that test always fails 🤔 It seems to run assertion as same as other tests, but assertion doesn't seem to run...

**Comment by ysugimoto** [comment]
Currently, network-error emulated test are skipping. I think it's enough for us.

**Comment by kdybicz** [comment]
yeah, I wanted to investigate on that, most of s3 tests doesn't seem to be executed. I was on holidays, so will try to catch up on that in the next couple of days it you're ok with that?

**Comment by ysugimoto** [comment]
Sure, thank you for that!

**Comment by kdybicz** [comment]
@ysugimoto if you're ok with my changes, branch should be ready to merge. Though I would suggest running tests locally few times on your machine before merging.

**Comment by ysugimoto** [comment]
@kdybicz Looks nice, thank you for fixing problem.I'll merge this PR 👍
